### DISCUSSION TITLE
Improve project maintainability.

### DISCUSSION THREAD

**Discussion by gibbz00** [helix-editor/helix#8358] [General]
~~Opening this as an issue for the lack of a better way to post this.~~

The goal of this thread is to stimulate an open discussion on how we can improve the workloads for those contributing to the project, hopefully reducing the issue and PR backlog. I love helix as a product, but have personally found contributing to it a bit of a disappointment. That is not in any way meant as an offense towards the developers, but more so an acknowledgement of the possibly unforeseen explosion in this project's popularity :)

Here are some starter questions I have myself:

* Are there workloads the current maintainers have to repeatedly do that could be systematized? (Common support questions that could be added to the F.A.Q, common and generic PR review suggestions etc.)
* How could the maintainers' confidence in the validity of newcomers pull requests be improved? Are there for example parts of the codebase that lack good testing solutions?
* Would it be prudent to integrate with https://bors.rust-lang.org/?
* Should an RFC process be introduced for larger changes?
* Are the maintainers open to onboarding more people into the project? (Not necessarily people with merge access.) 
* Should perhaps the current maintainer roles be documented/expanded so that some tasks can safely be delegated?
* Should a dedicated effort be but into going over this repo's PRs and issues? Things to look for might be checking which can be closed, or the possibility of adding more labels to increase their visibility.

**Comment by chtenb** [comment]
I notice that ~40% of the open PRs have status `waiting-for-peer-review`, suggesting it's difficult to find second reviewers. Would it help to somehow enlarge the pool of qualified reviewers?

Or alternatively, would it be possible to increase the amount of time core developers are able to spend on PR reviewing through sponsorship?
Donations were shortly discussed at https://github.com/helix-editor/helix/issues/2220 and have no clear goal currently. If there were to be a steady stream of donations, perhaps some developers would be willing to work on Helix part-time as a job.

**Comment by gibbz00** [comment]
Using the sponsorship fund to help increase the amount of qualified reviewers is a good idea, so long as it is not abused, of course. Having someone that works on it as a part-time job can very well prevent this.

Onboarding is unfortunately hard, especially in open source, where devs often come and go without notice. Figuring out trust in new maintainers is far from trivial. @archseer should weigh in on what he tends to look for in maintainers. He's the one with the final say after all.

On another note, maybe a some sort roadmap/project board can be created? 
The aim would be to give everyone an overview on which parts are being worked on and which PRs are blocking progress.
